### DISCUSSION TITLE
김단우 - 다변량 시계열 예측 - 상점 신용카드 매출 예측

### DISCUSSION THREAD

**Discussion by DANU011** [dinnercoding/wanted-pre-onboarding-challenge-data#102] [2023년 11월 프리온보딩 데이터 챌린지]
# 도메인 선택 이유
 - 매출 예측은 계절, 휴일, 이벤트 등 다양한 요인에 영향을 받아 크게 변동할 수 있습니다. 계절성을 가지는 데이터와 그에 따른 예측 모델을 만드는데 관심이 있어서 이 도메인을 선정하게 되었습니다. 

# 구체적인 문제 정의 
 - 목표 : 상점의 향후 특정 기간 동안의 신용카드 총매출을 예측하는 모델 개발.
 - 문제점 : 
   - 다양한 영향 인자 : 계절, 휴일, 이벤트, 마케팅 활동, 경기 변동 등 여러 요인이 매출에 영향을 미침.
   - 다양한 상점 유형 : 상점은 특성에 따라 매출 패턴이 다를 수 있어, 일반화된 모델이 필요함.
- 해결책 :
  - 다변량 시계열 분석 : 다양한 영향 인자를 고려한 시계열 분석을 통해 매출의 패턴을 파악하고 예측에 반영.
  - 상점 유형에 따른 개별 모델 : 상점 유형의 특성을 고려하여 개별 상점에 맞는 예측 모델을 구축하여 세분화된 예측 수행.

# 데이터 업로드
 상점 신용카드 매출 예측 경진대회 데이터 (출처 : DACON)

**Comment by ghost** [comment]
단우님! 상점 신용카드 총매출을 예측하는 모델을 개발한다는 목표는 잘 알겠지만 문제점에 대해 조금 더 디테일한 설명이 있으면 좋을 것 같아요~ 다양한 영향 인자와 다양한 상점 유형은 모델링을 할 때 발생하는 문제로 볼 수는 있겠지만 근본적인 문제점은 아니라는 생각이 드네요. 총매출을 예측하는 모델을 개발하면 어떤 문제를 해결할 수 있고 그로 인해 어떤 이점을 얻을 수 있는지에 대한 부분이 추가되면 좋을 것 같습니다 :) 혹시 제가 잘못 생각하고 있는 거라면 말씀해주세요!

**Reply by DANU011** [reply]
말씀해 주신 내용을 주의 깊게 살펴보았습니다. 감사합니다. 언급해 주신 부분에 대해서 저도 동의합니다. 처음에는 주제만 보고 아래와 같이 작성했었는데, 대회 배경을 보니 상점이 데이터를 분석을 통해 비즈니스에 활용한다기보다는 금융 회사가 상환 기간의 매출을 예측하여 신용 점수가 낮거나 담보를 가지지 못하는 우수 상점들에 금융 기회를 제공하려는 목적이라 데이터의 특성만 학습하기위해 현재 업로드되어 있는 내용으로 작성하였습니다. 대회의 목적에 따라 미래의 매출을 예측하여 신용 점수가 낮거나 담보를 가지지 못하는 우수 상점들에 금융 기회를 제공한다는 내용을 기반으로 목표, 문제점, 해결책을 보완하면 될 것 같습니다. 혹시 다른 의견이 있으시면 코멘트 부탁드립니다!

아래는 처음에 작성했던 내용입니다.

비즈니스 이점:
이 모델은 신용카드 총 매출을 더 정확하게 예측함으로써 상점 소유자 및 기타 이해관계자들에게 다양한 비즈니스적 가치를 제공할 것입니다. 정확한 매출 예측은 비즈니스 전략의 기반을 형성하며, 이는 상점 운영 및 마케팅 활동을 최적화하는 데 큰 도움이 될 것입니다. 상점 소유자는 재고 관리 및 직원 일정을 효과적으로 계획할 수 있고, 마케팅 팀은 특정 이벤트나 프로모션에 대한 전략을 세우는 데 더 정확한 정보를 활용할 수 있을 것입니다.

문제 해결:
다양한 영향 인자와 상점 유형을 고려하는 이유는 다양성을 고려하여 매출 예측의 정확성을 높이고자 합니다. 계절, 휴일, 이벤트, 마케팅 활동, 경기 변동 등 다양한 영향 인자를 고려함으로써 발생할 수 있는 변화무쌍한 상황에 유연하게 대응할 수 있을 것입니다. 또한, 상점 유형에 따라 매출 패턴이 다를 수 있기 때문에, 이를 반영하여 더 세분화된 예측을 수행하는 것이 중요합니다. 이는 각 상점이 특정 환경에서 직면하는 독특한 변화무쌍한 대한 효과적인 대응을 가능케 할 것입니다.

이해관계자에 대한 이점:
이 모델은 다양한 이해관계자들에게 실질적인 가치를 제공합니다. 상점 소유자들은 정확한 매출 예측을 통해 비즈니스 운영에 대한 투명성을 높이고 자금을 효율적으로 관리할 수 있습니다. 또한, 운영 일정과 공급망을 최적화하여 생산성을 향상시킬 수 있습니다. 마케팅 팀은 더 정확한 예측을 기반으로 타겟 마케팅 전략을 개발하고 성과를 최적화할 수 있습니다. 또한, 재무팀은 예산을 더 효율적으로 할당하고 재무 계획을 수립함으로써 기업의 금융 안정성을 강화할 수 있습니다. 이러한 혜택들은 기업 전반에 걸쳐 생산성을 향상시키며, 이해관계자들이 더 높은 수준의 전략적 의사결정을 내릴 수 있도록 도와줍니다.

모델의 활용 방안:
개발된 모델은 실시간 예측을 통해 상점이 빠르게 변화하는 환경에 신속히 대응할 수 있을 것입니다. 또한, 향후 몇 달 동안의 전략을 수립하는 데 사용될 수 있어, 기업은 장기적인 비전을 수립하고 이에 따른 효율적인 리소스 할당이 가능할 것입니다. 이는 상점의 경쟁력을 향상시키고 지속적인 성장을 이끌어낼 수 있을 것입니다.

**Reply by ghost** [reply]
오 좋습니다! 문제가 훨씬 명확해졌네요~ 정확한 매출을 예측함으로써 여러 관점에서 봤을 때 이점들을 잘 정리해주셨네요. 
신용 점수가 낮거나 담보를 가지지 못하지만 미래 매출이 우수할 것으로 기대되는 상점들은 어떠한 특징이 있는지 시각화해보시면서 
관련있는 인자는 무엇이고 관련없는 인자는 무엇일지 고민해보며 머신러닝 모델링을 해보시면 좋을 것 같습니다!
